src/tools/scripts/run_linter_locally.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+######################################
+## Custom Web Almanac script        ##
+######################################
+#
+# Run the GitHub Super-Linter locally
+# useful to avoid waiting on the push to get feed back
+# 
+# It can be run to lint everthing if no params are passed
+# Or pass one or more params to just lint those files or folders
+#
+# Note this script should only be called from the super-linter Docker container and not directly
+#
+
+# exit when any command fails instead of trying to continue on
+set -e
+
+# This script must be run from src directory
+if [ -d "src" ]; then
+  cd src
+fi
+
+# Annoyingly super-linter includes node_modules and env which take a long time
+# https://github.com/github/super-linter/issues/985
+# So let's copy to /tmp folder and lint from there. It has the added advantage
+# of us being able to lint specific subsets of files easily.
+
+# Remove old files if they exist
+rm -rf /tmp/lint
+mkdir /tmp/lint/
+
+echo "Copying files to /tmp"
+# Check if provided files on command line or linting all files
+if [ "$#" -gt 0 ]; then
+  # Copy linter config files
+  mkdir /tmp/lint/.github
+  cp -r ../.github/linters /tmp/lint/.github
+  # Copy files provided
+  for FILES in "$@"
+  do
+    echo "- Copying ${FILES}"
+    cp -r "${FILES}" "/tmp/lint/"
+  done
+else
+  echo "Copying all files"
+  # Copy everything except node_folders and env
+  cp -r ../.github /tmp/lint
+  mkdir /tmp/lint/src
+  find ./ -maxdepth 1 -not -name "." -not -name "node_modules" -not -name "env" -exec cp -r {} /tmp/lint/src/ \;
+  # Delete some large files
+  rm -rf /tmp/lint/src/static/fonts
+  rm -rf /tmp/lint/src/static/images
+  rm -rf /tmp/lint/src/static/pdfs
+fi
+
+# set all the necessary environments variables
+export RUN_LOCAL=true
+export VALIDATE_BASH=true
+export VALIDATE_CSS=true
+export VALIDATE_HTML=true
+export VALIDATE_JAVASCRIPT_ES=true
+export VALIDATE_JSON=true
+export VALIDATE_MD=true
+export VALIDATE_PYTHON_PYLINT=true
+export VALIDATE_PYTHON_FLAKE8=true
+export VALIDATE_YAML=true
+
+echo "Starting linting"
```

Comment:
How does the linter know that want it to lint files under the `/tmp/` folder and not under the `app` folder? Do we need to change the working directory to `/tmp/lint/src`?

Reply:
The Linter always lints files under `/tmp/lint`. Normally you would map that directly to your git directory (which is what I originally had), but it's only because we override that that this is even an issue.

Also note the Linter needs to be able to see the `.github` directory to get our linter rule configurations so definitely can't change it to `src`.

Reply:
Got it! I didn't know they have a magic location in place as the default. It perhaps is a good idea that linting is happening elsewhere, in case the process yields any files or changes the state of existing ones. Especially, when running in Docker and proper UID mapping is not in place, the process may create files under the mounted folder with root permissions.

Reply:
Yeah will you'd think. But not sure it's that safe - I managed to delete my whole repo by mapping `/tmp/lint/` to the actual directory and then my script deleted that - oops! Ah well was time for a clean up anyway and everything was checked in on GitHub so luckily didn't lose anything!

 So just because it's in Docker doesn't mean it can't delete or overwrite the files. Still got to trust what we're doing, when adding this to our toolset (and it's already running in a GitHub Action). Plus that's what Git is for - to store the main copy offline!